### PR TITLE
Add checkpoints for Terraform codelab

### PR DESCRIPTION
**Pull request by rainshen49**
Add sample Terraform files for FriendlyChat

### CODE REVIEW

**Comment by rainshen49** [comment]
@mraouffouad PTAL

**Comment by rainshen49** [comment]
cc @tylerg-dev

File: web/terraform-checkpoints/replicate-config/main-foreach.tf [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+resource "google_firebase_web_app" "default" {
+  provider = google-beta
+  for_each = google_firebase_project.default
+
+  project         = each.value.project
+  display_name    = "<DISPLAY_NAME_OF_YOUR_WEB_APP>"
+  deletion_policy = "DELETE"
+}
+
+# Uncomment below if you set up Firebase Authentication using Terraform in the previous step
```
Comment (mraouffouad) [inline web/terraform-checkpoints/replicate-config/main-foreach.tf:91]: I would put these kinds of instructional comments in all-caps.

Comment (rainshen49) [reply web/terraform-checkpoints/replicate-config/main-foreach.tf]: Done.

File: web/terraform-checkpoints/create-firebase-project-checkpoint.tf [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+# Configure the provider that uses the new project's quota.
+provider "google-beta" {
+  user_project_override = true
+}
+
+# Create a new Google Cloud project.
+resource "google_project" "default" {
+  provider = google-beta.no_user_project_override
+
+  name       = "<PROJECT_NAME_OF_YOUR_PROJECT>"
```
Comment (mraouffouad) [inline web/terraform-checkpoints/create-firebase-project-checkpoint.tf:27]: What do you think about adding a comment (here or at the top of the resource) to highlight these parameters that need to be replaced and their targetted values?

**Review by mraouffouad — APPROVED** [review]

File: web/terraform-checkpoints/replicate-config/main-foreach.tf [hunk 3]
```diff
@@ -0,0 +1,325 @@
...
+# }
+
+# resource "google_identity_platform_default_supported_idp_config" "google_sign_in" {
+#   provider = google-beta
+#   for_each = google_firebase_project.default
+
+#   project       = each.value.project
+#   enabled       = true
+#   idp_id        = "google.com"
+#   # TODO: replace with your own values
```
Comment (mraouffouad) [inline web/terraform-checkpoints/replicate-config/main-foreach.tf:138]: nit: s/values/value

**Review by mraouffouad — APPROVED** [review]

File: web/terraform-checkpoints/create-firebase-project-checkpoint.tf (hunk 2 above)
Comment (mraouffouad) [reply web/terraform-checkpoints/create-firebase-project-checkpoint.tf:28]: I would also write these TODOs in all-caps.

File: web/terraform-checkpoints/create-firebase-project-checkpoint.tf [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  project_id = "<PROJECT_ID_OF_YOUR_PROJECT>"
+
+  # Required for the project to display in any list of Firebase projects.
+  labels = {
+    "firebase" = "enabled"
+  }
+}
+
+# Enable the required underlying Service Usage API.
```
Comment (tylerg-dev) [inline web/terraform-checkpoints/create-firebase-project-checkpoint.tf:37]: Do we really want the code-lab checkpoints to have the 'serviceusage' blocks to be unrolled?  it's a lot more compact to use the loops & just add new rows.

Comment (mraouffouad) [reply web/terraform-checkpoints/create-firebase-project-checkpoint.tf:37]: I think we had to flatten it this way so it doesn't conflict with the for_each used in replicating the config across various envs (main-foreach.tf) below.

Comment (rainshen49) [reply web/terraform-checkpoints/create-firebase-project-checkpoint.tf:37]: Yes, if we use for_each on services, we end up needing a double for-loop for the combo of (prod, staging) and (serviceusage, firebase) for the project replication step.

File: web/terraform-checkpoints/create-firebase-project-checkpoint.tf (hunk 2 above)
Comment (rainshen49) [reply web/terraform-checkpoints/create-firebase-project-checkpoint.tf:28]: Done.

File: web/terraform-checkpoints/replicate-config/main-foreach.tf (hunk 3 above)
Comment (rainshen49) [reply web/terraform-checkpoints/replicate-config/main-foreach.tf]: Done

**Pull request merged**